pulsar-metadata/src/test/java/org/apache/pulsar/metadata/MetadataStoreTest.java

```diff
@@ -500,6 +507,35 @@ public void testThreadSwitchOfZkMetadataStore(boolean hasSynchronizer, boolean e
         }).join();
     }
 
+    @Test
+    public void testZkLoadConfigFromFile() throws Exception {
+        final String metadataStoreName = UUID.randomUUID().toString().replaceAll("-", "");
+        MetadataStoreConfig.MetadataStoreConfigBuilder builder =
+                MetadataStoreConfig.builder().metadataStoreName(metadataStoreName);
+        builder.fsyncEnable(false);
+        builder.batchingEnabled(true);
+        builder.configFilePath("src/test/resources/zk_client_disabled_sasl.conf");
+        MetadataStoreConfig config = builder.build();
+        @Cleanup
+        ZKMetadataStore store = (ZKMetadataStore) MetadataStoreFactory.create(zks.getConnectionString(), config);
+
+        PulsarZooKeeperClient zkClient = (PulsarZooKeeperClient) store.getZkClient();
+        assertFalse(zkClient.getClientConfig().isSaslClientEnabled());
+
+        zkClient.process(new WatchedEvent(Watcher.Event.EventType.None, Watcher.Event.KeeperState.Expired, null));
+
+        Awaitility.await().untilAsserted(() -> {
+            AtomicReference<ZooKeeper> zk =
+                    (AtomicReference<ZooKeeper>) FieldUtils.readDeclaredField(zkClient, "zk", true);
+            assertNotNull(zk.get());
+        });
+        AtomicReference<ZooKeeper> zk =
+                (AtomicReference<ZooKeeper>) FieldUtils.readDeclaredField(zkClient, "zk", true);
+        assertNotNull(zk.get());
+        ZooKeeper zooKeeper = zk.get();
+        assertFalse(zooKeeper.getClientConfig().isSaslClientEnabled());
```

Comment:
Nit. This looks cleaner:

```suggestion
        var zooKeeperRef = (AtomicReference<ZooKeeper>) WhiteboxImpl.getInternalState(zkClient, "zk");
        var zooKeeper = Awaitility.await().until(zooKeeperRef::get, Objects::nonNull);
        assertFalse(zooKeeper.getClientConfig().isSaslClientEnabled());
```